app/services/puppet_fact_parser.rb

```diff
@@ -140,6 +140,14 @@ def cores
     facts.dig('processors', 'count') || facts['processorcount']
   end
 
+  def foreman_uuid
+    facts['foreman_uuid']
+  end
+
+  def smart_proxy_uuid
+    facts['smart_proxy_uuid']
+  end
+
```

Comment:
I'd prefer structured facts, meaning `foreman::server_uuid` or something like that.

Reply:
That would be slightly tricky. if we deployed static facts the way we do know. It could be doable if we switched to using dynamic facts on the puppet side as @ewoud suggested in one of the other prs. I'll take this into consideration

Reply:
Oh right I remember that was painful. Okay.